mdc/report/report.py

```diff
@@ -560,18 +579,23 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None,
                    orderby=False, lazy=True):
         res = super(RptCumulative, self).read_group(domain, fields, groupby,
                                                     offset=offset, limit=limit, orderby=orderby, lazy=lazy)
-        if 'quality' in fields or 'total_yield' in fields:
+        # we don´t calculate ind_ fields because they aren´t in tree view
+        # group_fields = ['quality', 'total_yield', 'ind_backs', 'ind_mo', 'ind_crumbs', 'ind_quality','ind_cleaning']
+        group_fields = ['quality', 'total_yield']
+        if any([x in fields for x in group_fields]):
             for line in res:
                 if '__domain' in line:
                     quality_weight = 0
-                    total_yield_weight = 0
                     total_weight = 0
+                    total_yield_weight = 0
+                    total_gross_weight = 0
                     lines = self.search(line['__domain'])
                     for line_item in lines:
-                        quality_weight += line_item.quality * line_item.product_weight
-                        total_yield_weight += line_item.total_yield * line_item.product_weight
+                        quality_weight += line_item.quality * (line_item.product_weight + line_item.shared_product_weight / 2)
                         total_weight += line_item.product_weight + line_item.shared_product_weight / 2
+                        total_yield_weight += line_item.product_weight  + line_item.sp1_weight
+                        total_gross_weight += line_item.gross_weight
                     if total_weight > 0:

```

Comment:
Prevent dividing by zero

```suggestion
                    if total_weight > 0 and total_gross_weigh > 0:

```

Reply:
Ok. I think i better put the "if total_gross_weight > 0:" only for total_yield calculate:
                    if total_weight > 0:
                        line['quality'] = quality_weight / total_weight
                    if total_gross_weight > 0:
                        line['total_yield'] = total_yield_weight / total_gross_weight

Reply:
I agree

Reply:
Thanks for your suggestion anyway, it made me see an improvement there.

Reply:
I think that additional `100` is a deliberated change, isn't it?

Reply:
Yes, When I checked that code at your request, I saw that this value had to be percentage